mypy/specialtype.py

```diff
@@ -0,0 +1,912 @@
+from collections import OrderedDict
+
+from typing import List, Dict, Tuple, cast, Optional, Callable, TYPE_CHECKING
+
+from mypy.exprtotype import expr_to_unanalyzed_type, TypeTranslationError
+from mypy.nodes import (
+    TypeInfo, AssignmentStmt, FuncDef, ClassDef, Var, GDEF, Expression,
+    Block, NameExpr, TupleExpr, ListExpr, ExpressionStmt, PassStmt,
+    DictExpr, CallExpr, RefExpr, Context, SymbolTable, UNBOUND_TVAR,
+    MDEF, Decorator, TypeVarExpr, NewTypeExpr, StrExpr, BytesExpr,
+    ARG_POS, ARG_NAMED, ARG_NAMED_OPT, NamedTupleExpr, TypedDictExpr, Argument,
+    UnicodeExpr, EllipsisExpr, TempNode, EnumCallExpr,
+    COVARIANT, CONTRAVARIANT, INVARIANT, ARG_OPT, SymbolTableNode
+)
+from mypy.types import (
+    NoneTyp, CallableType, Instance, Type, TypeVarType, AnyType,
+    TypeVarDef, TupleType, UnboundType, TypedDictType, TypeType,
+)
+from mypy import join
+if TYPE_CHECKING:
+    import mypy.semanal
+
+
+class Special:
+    """
```

Comment:
Style nit: Add summary of what this class does as the first line of the docstring, followed by an empty line.